### DISCUSSION TITLE
{{ layout }} not showing?

### DISCUSSION THREAD

**Discussion by stephenmeehanuk** [statamic/cms#7615] [Troubleshooting] [answered]
Hi

I often use this to show what template is in use during development

```
{{ if environment === 'local' }}
{{ current_template }}
{{ /if }}
```

I thought it'd be useful to also output the `{{layout}}` too

But it doesn't seem to work everywhere?

```
{{ if environment === 'local' }}
{{ current_template }}
{{ layout }}
{{ /if }}
```

For example `pages.yaml` contains this

```
title: Pages
template: pages/show
layout: layouts/triple
```

But when I visit a page, even after `php artisan optimize:clear` my `{{ layout }}` is blank?

I've tried creating a new page, `{{ current_template }}` shows but `{{ layout }}` remains blank.

Both `{{ current_template }}` and `{{ layout }}`  work when I visit the home page.

Any ideas?

```
Environment
Application Name: Statamic
Laravel Version: 9.52.4
PHP Version: 8.2.2
Composer Version: 2.4.1
Environment: local
Debug Mode: ENABLED
URL: statamic-bones-tw.test
Maintenance Mode: OFF

Cache
Config: NOT CACHED
Events: NOT CACHED
Routes: NOT CACHED
Views: CACHED

Drivers
Broadcasting: log
Cache: redis
Database: mysql
Logs: stack / single
Mail: smtp
Queue: redis
Session: file

Statamic
Addons: 1
Antlers: runtime
Stache Watcher: Enabled
Static Caching: Disabled
Version: 3.4.4 PRO

Statamic Addons
simonhamp/statamic-heroicons: 1.1.0
```

**Comment by duncanmcclean** [comment]
On the page where it's not working, is that a custom route or an entry page?

**Reply by stephenmeehanuk** [reply]
It's an entry page. It's just a basic entry in the pages collection. No custom routes.

**Reply by jasonvarga** [reply]
Does your `home` page have `layout` in the yaml?

**Reply by stephenmeehanuk** [reply]
It does, yes.

```
id: home
blueprint: home
title: Home
layout: layouts/layout
template: templates/home
```

**Comment by jasonvarga** [comment, marked as answer]
We don't provide the `layout` variable explicitly. You're only getting it on entry that actually have a `layout` variable in their data. (Same thing happens with `template`)

We could provide a `current_layout`, like we do with `current_template`. No one had ever asked for it before. 😄

**Reply by jasonvarga** [reply]
See #7623

**Reply by stephenmeehanuk** [reply]
Thanks for letting me know. Does this mean `current_layout` will be available in a future release?

**Reply by jasonvarga** [reply]
Yes
